guac_core/src/channel_client/channel_manager.rs

```diff
@@ -92,29 +135,38 @@ impl ChannelManager {
                     state: state.clone(),
                 }
             }
-            _ => bail!("can only propose if not accepted and in state Proposed"),
+            ChannelManager::Proposed {
+                accepted: true,
+                state,
+            } => {
+                ret = ChannelManagerAction::SendNewChannelTransaction(state.clone());
+                ChannelManager::PendingCreation {
+                    state: state.clone(),
+                    pending_send: 0.into(),
+                }
+            }
+            _ => bail!("can only propose if in state Proposed"),
         };
 
         Ok(ret)
     }
 
-    fn channel_created(
+    pub fn channel_created(
         &mut self,
         channel: &Channel,
         our_address: EthAddress,
-    ) -> Result<ChannelManagerAction, Error> {
-        let ret;
+    ) -> Result<(), Error> {
         *self = match self {
             ChannelManager::Proposed { .. } | ChannelManager::PendingCreation { .. } => {
                 // TODO: verify it actually made it into the blockchain
                 if is_channel_acceptable(&channel)? {
                     if channel.address_a == our_address {
-                        ret = ChannelManagerAction::None;
+                        // we created this transaction
                         ChannelManager::Joined {
                             state: CombinedState::new(&channel),
                         }
                     } else {
-                        ret = ChannelManagerAction::None;
+                        // They created this transaction
```

Comment:
You're assuming a different address necessarily means they've created the channel. What if for whatever reason gibberish finds its way into `channel.address_a`? Is there a way to `else if` and check against `theirs` too?